### PR TITLE
Handling of compressed "body" in logging middleware.

### PR DESCRIPTION
**Pull request by peterschutt**
If a compression middleware is configured, upon compression of a response we add a flag to connection scope. Logging middleware checks that flag to determine how logging of "body" should be handled.

Adds `include_compressed_body` to `LoggingMiddlewareConfig`.

If response body is compressed and `include_compressed_body` is `True` we do not decode the body. If `False` (the default), body is excluded from log.

Introduces a namespace for storing internal state in connection scope, and utils `get_starlite_scope_state` and `set_starlite_scope_state`.

# PR Checklist

- [x] Have you followed the guidelines in `CONTRIBUTING.md`?
- [x] Have you got 100% test coverage on new code?
- [x] Have you updated the prose documentation?
- [x] Have you updated the reference documentation?

### CODE REVIEW

File: starlite/utils/scope.py [hunk 1]
```diff
@@ -19,3 +21,28 @@ def get_serializer_from_scope(scope: "Scope") -> Optional["Serializer"]:
...
     app = scope["app"]
     return app.response_class.serializer if app.response_class else None
+
+
+def get_starlite_scope_state(scope: "Scope", key: str, default: Any = None) -> Any:
+    """Get an internal value from connection scope state.
+
+    Args:
+        scope: The connection scope.
+        key: Key to set under internal namespace in scope state.
```
Comment (peterschutt) [inline starlite/utils/scope.py:31]:
```suggestion
        key: Key to get from internal namespace in scope state.
```

File: starlite/utils/scope.py [hunk 2]
```diff
@@ -19,3 +21,28 @@ def get_serializer_from_scope(scope: "Scope") -> Optional["Serializer"]:
...
+        default: Value set in internal namespace and returned if `key` doesn't exist.
+
+    Returns:
+        Value mapped to `key` in internal connection scope namespace. Returns `None` if `key` not in internal namespace.
```
Comment (peterschutt) [inline starlite/utils/scope.py:35]:
```suggestion
        Value mapped to `key` in internal connection scope namespace. Returns `default` if `key` not in internal namespace.
```

File: starlite/constants.py [hunk 3]
```diff
@@ -1,8 +1,13 @@
...
+EXTRA_KEY_REQUIRED = "required"
 EXTRA_KEY_SKIP_VALIDATION = "skip_validation"
+EXTRA_KEY_VALUE_TYPE = "value_type"
+HTTP_RESPONSE_START = "http.response.start"
+HTTP_RESPONSE_BODY = "http.response.body"
+REDIRECT_STATUS_CODES = {301, 302, 303, 307, 308}
+RESERVED_KWARGS = {"state", "headers", "cookies", "request", "socket", "data", "query", "scope"}
+SCOPE_STATE_DEPENDENCY_CACHE = "dependency_cache"
+SCOPE_STATE_RESPONSE_COMPRESSED = "response_compressed"
+STARLITE = "__starlite__"
```
Comment (Goldziher) [inline starlite/constants.py:13]: i would rename the constant from `STARLITE` into something like `STARLITE_RESERVED_STATE_NAMESPACE` or something.

**Review by Goldziher — APPROVED** [review]
cool

**Pull request merged**